types/react-relay/index.d.ts

```diff
@@ -76,7 +77,7 @@ export const graphql: GraphqlInterface;
 export interface QueryRendererProps {
     cacheConfig?: RelayRuntimeTypes.CacheConfig;
     environment: RelayRuntimeTypes.Environment;
-    query: RelayRuntimeTypes.GraphQLTaggedNode;
+    query: RelayRuntimeTypes.GraphQLTaggedNode | null;
```

Comment:
This still requires you to explicitly specify `null`, so it should probably be

```ts
query?: RelayRuntimeTypes.GraphQLTaggedNode | null;
```

(I notice the `render` prop specifies `undefined` rather than using `?`, mind fixing that as well while you’re at it?)

Reply:
I also did not realize this was optional

Reply:
@alloy Yes, it looks as though from the `react-relay` code that `query` is only checked for truthiness, so I'll make your requested change.

@voxmatt it's quite a handy pattern when doing something like optionally sending a search request if your query is truthy.